### DISCUSSION TITLE
Stop replacing spaces in enum values with underscores

### DISCUSSION THREAD

**Discussion by devuxer** [prisma/prisma#5881] [Q&A] [answered]
I have this in my schema.prisma:

```prisma
enum Result {
  Complete
  No_Show @map("No Show")
  Walkover
  Retired
  No_Reply @map("No Reply")
  Unavailable
}
```

It appears if I do a query, Prisma sticks underscores into my enum values, and when I mutate, Prisma removes the underscores from my enum values.

This is breaking my system. My enum values are intended to be human readable, which allows me to use them to generate select lists, radio buttons, etc. I don't want users to see underscores. There is no technical reason why enum values can't have spaces (JS, TS, and Postgres all support them), so why does Prisma apply this artificial limitation?

Perhaps a way to allow this without any breaking changes would be to simply allow enum values to be quoted within prisma.introspect.

**Comment by ryands17** [comment, marked as answer]
Hey @devuxer 👋 

Unfortunately having values with spaces is not possible in enums. We have a request [here](https://github.com/prisma/prisma/issues/4954) for the same and it would be great if you could add a 👍  along with your use case so that we can look into this.

**Reply by devuxer** [reply]
Thanks for the link. Somehow missed that while searching. I upvoted and added a comment.

**Comment by maitrungduc1410** [comment]
I'm facing same issue, when running `migrate`, enum value for my column in my database has space (expected), but the enum value generated (in `node_modules`) by prisma is tranformed to be underscore. it's inconsistent

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
